url0067cdd33d_goldolphin_Mi_tgz-pJ8-mi_regex_CommonJ8/src/mi/legacy/parser/Parser.java

```diff
@@ -0,0 +1,134 @@
+package mi.legacy.parser;
+
+import mi.stream.ICharStream;
+
+import java.util.ArrayList;
+
+/**
+ * @author goldolphin
+ *         2014-02-13 20:30
+ */
+public class Parser {
+    private final Nonterm[] nontermTable;
+    private final State startState;
+    private final int topHeadId;
+    private final boolean useHeadSet;
+
+    Parser(Nonterm[] nontermTable, State startState, int topHeadId, boolean useHeadSet) {
+        this.nontermTable = nontermTable;
+        this.startState = startState;
+        this.topHeadId = topHeadId;
+        this.useHeadSet = useHeadSet;
+    }
+
+    public boolean parse(ICharStream stream) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 15. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.